networkx/algorithms/distance_measures.py

```diff
@@ -11,10 +11,182 @@
+    
+    Computation is based on smart lower and upper bounds, and in practice 
+    linear in the number of nodes, rather than quadratic (except for some 
+    border cases such as complete graphs or circle shaped graphs). 
+
+    Parameters
+    ----------
+    G : NetworkX graph
+       An undirected graph
+
+    compute : string denoting the requesting metric; 
+       "diameter" for the maximal eccentricity value, 
+       "radius" for the minimal eccentricity value,  
+       "periphery" for the set of nodes with eccentricity equal to the diameter,
+       "center" for the set of nodes with eccentricity equal to the radius       
+
+    Returns
+    -------
+    value : value of the requested metric;
+       int for "diameter" and "radius" or 
+       list of nodes for "center" and "periphery"
+       
+    Raises
+    ------
+    NetworkXError
+        If the graph consists of multiple components
+       
+    Notes
+    -----       
+    This algorithm was proposed in the following papers:
+    
+    F.W. Takes and W.A. Kosters, Determining the Diameter of Small World 
+    Networks, in Proceedings of the 20th ACM International Conference on 
+    Information and Knowledge Management (CIKM 2011), pp. 1191-1196, 2011. 
+    doi: http://dx.doi.org/10.1145/2063576.2063748
+    
+    F.W. Takes and W.A. Kosters, Computing the Eccentricity Distribution of 
+    Large Graphs, Algorithms 6(1): 100-118, 2013. 
+    doi: http://dx.doi.org/10.3390/a6010100
+    
+    M. Borassi, P. Crescenzi, M. Habib, W.A. Kosters, A. Marino and F.W. Takes, 
+    Fast Graph Diameter and Radius BFS-Based Computation in (Weakly Connected) 
+    Real-World Graphs, Theoretical Computer Science 586: 59-80, 2015. 
+    doi: http://dx.doi.org/10.1016/j.tcs.2015.02.033   
+    """
+
+    # init variables
+    degrees = dict(G.degree())  # start with the highest degree node
+    minlowernode = max(degrees, key=lambda i: degrees[i])
+    N=len(degrees)  # number of nodes
+    # alternate between smallest lower and largest upper bound
+    high = False
+    # status variables
+    ecc_lower = dict.fromkeys(G.nodes(), 0)
+    ecc_upper = dict.fromkeys(G.nodes(), N)
+    candidates = set(G)
+    
+    # (re)set bound extremes
+    minlower = N
+    maxlower = 0
+    minupper = N
+    maxupper = 0
+    
+    # repeat the following until there are no more candidates
+    while candidates:
+        if high:
+            current = maxuppernode; # select node with largest upper bound
+        else:
+            current = minlowernode; # select node with smallest lower bound
+        high = not high
+        
+        # get distances from/to current node and derive eccentricity
+        dist = dict(networkx.single_source_shortest_path_length(G, current))
+        if len(dist) != N:
+            msg = ('Cannot compute metric because graph is not connected.')
+            raise networkx.NetworkXError(msg)
+        current_ecc = int(max(dist.values()))
```

Comment:
The case in which infinite path lengths (because there is no path) could occur is already caught by the if()-check in line 93.
So, given that single_source_shortest_path_length() indeed always returns integers as stated in the function's specification, no, casting to int is not per se needed and could indeed be removed.